jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp

```diff
@@ -556,6 +556,11 @@ ecma_builtin_date_dispatch_construct (const ecma_value_t *arguments_list_p, /**<
 
   if (ecma_is_completion_value_empty (ret_value))
   {
+    if (!ecma_number_is_nan (*prim_value_num_p) && ecma_number_is_infinity (*prim_value_num_p))
```

Comment:
But this tests that if it is **not** NaN and is infinity.

Also note that in the first line of the `ecma_number_is_infinity` there is an assert:
```
JERRY_ASSERT (!ecma_number_is_nan (num));
```